include/private/hashmap.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Hash map support
+ *
+ * Copyright 2020 Philip Rebohle for Valve Corporation
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#ifndef __VKD3D_HASHMAP_H
```

Comment:
Is this implementation based on existing, well tested code? Not sure how detailed I should be in reviewing this.

Reply:
It's a custom implementation, and doesn't even have a `remove` function or anything like that because we're not using it.